Document IPI for vSphere w/ existing resource pool

willhaines opened the pull request:
Document required permissions and usage of IPI installation on vSphere with and existing resource pool. Note that this process allows a narrower scope of permissions in vSphere, which is very useful for users on shared vmware infrastructure.

fixes #43011

jinyunma commented on modules/installation-installer-provisioned-vsphere-config-yaml.adoc:
```diff
@@ -79,6 +79,7 @@ platform:
+    resourcePool: resource_pool
     diskType: thin <6>
     network: VM_Network
     cluster: vsphere_cluster_name <7>
```
I saw that here <7> explains the option of resourcePool, not cluster, need to move <7> to the end of option resourcePool.

willhaines replied:
@jinyunma Thanks for catching that. I think I've clarified the difference between resource pool and vSphere cluster now.

jinyunma commented:
The update looks good to me,  cc: @rvanderp3 for review if any more comments.

rvanderp3 commented:
/lgtm

jinyunma commented:
/lgtm

kalexand-rh commented:
@mjpytlak, will you PTAL?

bscott-rh commented:
@willhaines hello Will - please squash your commits down to a single commit per the [contribution guidelines](https://github.com/openshift/openshift-docs/blob/main/contributing_to_docs/create_or_edit_content.adoc#understanding-the-contribution-workflow). Instructions for squashing are [further down in the same document](https://github.com/openshift/openshift-docs/blob/main/contributing_to_docs/create_or_edit_content.adoc#add-changes-to-an-existing-pr-if-required). Thank you

willhaines commented:
Hi @bscott-rh thanks for the feedback. I've squashed my commits and also went ahead and rebased onto the latest main. Luckily, the rebase did not cause any conflicts.

willhaines commented:
Thanks @mjpytlak, I committed all of your suggestions, then squashed the additional commit. Just for my own education, is [this](https://redhat-documentation.github.io/supplementary-style-guide/) the style guide that you referred to?

mjpytlak commented:
> Just for my own education, is [this](https://redhat-documentation.github.io/supplementary-style-guide/) the style guide that you referred to?

Yes. And an IBM style guide that, unfortunately, is not public. I will ask someone on my team to merge your changes. Thank you again for taking the time to contribute to the doc set.

mjpytlak commented:
Doc peer review LGTM.

mjpytlak commented:
@jinyunma Ahead of a merge, can you confirm which OpenShift releases this update applies to, so I can make sure it gets CP into the correct doc branches. Thanks!

mjpytlak commented:
Peer review squad: Please CP to 4.10 and 4.11.

jeana-redhat commented:
/cherrypick enterprise-4.11

jeana-redhat commented:
/cherrypick enterprise-4.10

jeana-redhat commented:
@willhaines thanks so much for this excellent contribution! :tada: Once the cherry picks build I will merge them. You should see you change live later today :slightly_smiling_face:
